.gitignore

```diff
@@ -9,4 +9,9 @@ Cargo.lock
 # These are backup files generated by rustfmt
 **/*.rs.bk
 
-.venv
\ No newline at end of file
+.venv
+
+.github
+.cargo
+.idea
+/docs
```

Comment:
I'm also wondering if we should check in docs as well. In the future I'd imagine we can have stuff like architecture.md in there.